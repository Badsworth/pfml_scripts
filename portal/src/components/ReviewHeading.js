import Heading from "./Heading";
import Link from "next/link";
import PropTypes from "prop-types";
import React from "react";

/**
 * Heading for a group of ReviewRow components
 */
const ReviewHeading = (props) => {
  return (
    <div className="display-flex flex-align-end margin-top-6 margin-bottom-3">
      <Heading
        className="flex-fill margin-right-1"
        id={props.id}
        level={props.level}
        size="2"
      >
        {props.children}
      </Heading>
      {props.editHref && (
        <Link href={props.editHref}>
          <a
            className="usa-link margin-0"
            aria-label={`${props.editText}: ${props.children}`}
          >
            {props.editText}
          </a>
        </Link>
      )}
    </div>
  );
};

ReviewHeading.propTypes = {
  /**
   * Heading text
   */
  children: PropTypes.string.isRequired,
  /**
   * HTML `href` attribute for the edit link.
   * If undefined, no edit link will be shown.
   */
  editHref: PropTypes.string,
  /**
   * Localized text for the edit link
   */
  editText: PropTypes.node,
  // TODO this
  id: PropTypes.string,
  /**
   * The heading level to use
   */
  level: PropTypes.oneOf(["2", "3", "4", "5", "6"]).isRequired,
};

export default ReviewHeading;
