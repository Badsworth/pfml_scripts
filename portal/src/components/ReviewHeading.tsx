import Heading from "./core/Heading";
import Link from "next/link";
import React from "react";

interface ReviewHeadingProps {
  children: string;
  editHref?: string;
  editText?: React.ReactNode;
  level: "2" | "3" | "4" | "5" | "6";
}

/**
 * Heading for a group of ReviewRow components
 */
const ReviewHeading = (props: ReviewHeadingProps) => {
  return (
    <div className="display-flex flex-align-end margin-top-6 margin-bottom-3">
      <Heading
        className="flex-fill margin-right-1"
        level={props.level}
        size="2"
      >
        {props.children}
      </Heading>
      {props.editHref && (
        <Link href={props.editHref}>
          <a
            className="usa-link margin-0"
            aria-label={`${props.editText}: ${props.children}`}
          >
            {props.editText}
          </a>
        </Link>
      )}
    </div>
  );
};

export default ReviewHeading;
