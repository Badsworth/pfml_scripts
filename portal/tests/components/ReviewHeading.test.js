import { render, screen } from "@testing-library/react";
import React from "react";
import ReviewHeading from "../../src/components/ReviewHeading";

const renderComponent = (customProps) => {
  const props = {
    children: "Who is taking leave?",
    level: "2",
    ...customProps,
  };
  return render(<ReviewHeading {...props} />);
};

describe("ReviewHeading", () => {
  it("renders a Heading, no link if editHref not defined", () => {
    const { container } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.queryByRole("link")).not.toBeInTheDocument();
  });

  it("when editHref is defined renders with an edit link", () => {
    const { container } = renderComponent({
      editHref: "/name",
      editText: "Edit",
    });
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.queryByRole("link")).toBeInTheDocument();
  });
});
