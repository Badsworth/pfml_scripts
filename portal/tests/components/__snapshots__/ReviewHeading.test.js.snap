// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewHeading renders a Heading, no link if editHref not defined 1`] = `
<div
  class="display-flex flex-align-end margin-top-6 margin-bottom-3"
>
  <h2
    class="flex-fill margin-right-1 font-heading-md text-bold"
  >
    Who is taking leave?
  </h2>
</div>
`;

exports[`ReviewHeading when editHref is defined renders with an edit link 1`] = `
<div
  class="display-flex flex-align-end margin-top-6 margin-bottom-3"
>
  <h2
    class="flex-fill margin-right-1 font-heading-md text-bold"
  >
    Who is taking leave?
  </h2>
  <a
    aria-label="Edit: Who is taking leave?"
    class="usa-link margin-0"
    href="/name"
  >
    Edit
  </a>
</div>
`;
